zeq/eventDescriptor.h

```diff
@@ -32,7 +32,10 @@ enum EventDirection
  * It contains the REST api name, the 128bit zeq event id and the schema
  * needed for automatic JSON serialization/deserialization.
  */
-struct EventDescriptor : public boost::noncopyable
+struct EventDescriptor
+#ifndef BOOST_NO_CXX11_RVALUE_REFERENCES
+    : public boost::noncopyable
+#endif
```

Comment:
That's only true if the copy constructor has the non-const overload. However I'm wondering if that can cause problems in client code.

I remember playing with const and non-const copy constructors in the past and I remember having problems, but I can't tell what exactly right now.

Another alternative suggested at the bottom of [this page](http://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Move_Constructor), suggests to use a mutable member to pass the ownership from rhs. This solution needs the class to be copyable and allows data sharing between the multiple copies but doesn't need any const cast. I actually prefer this solution, despite being less safe, because we should consider the C++03 support a legacy feature.